addons/website_livechat/models/website_visitor.py

```diff
@@ -92,6 +93,17 @@ def action_send_chat_request(self):
                 notifications.append([(self._cr.dbname, 'res.partner', operator.partner_id.id), mail_channel_info])
             self.env['bus.bus'].sendmany(notifications)
 
+    def _link_to_visitor(self, target, keep_unique=True):
+        """ Copy sessions of the secondary visitors to the main partner visitor. """
+        if target.partner_id:
+            main_partner_visitor = self.env['website.visitor'].sudo().search(
```

Comment:
Actually main_partner_visitor should be target. If I remember this method, target is the one where we merge data, so normally no need to re-compute something here.